### DISCUSSION TITLE
[Feature]: How to stop Pinch Zoom / Resize of an object on touch screen

### DISCUSSION THREAD

**Discussion by Saqib92** [fabricjs/fabric.js#9347] [Q&A]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have searched and referenced existing issues, feature requests and discussions
- [X] I am filing a **FEATURE** request.

### Description

is there an option when creating an object for example:
```
new fabric.Line(points, {
            strokeWidth: this.lineSize,
            fill: this.drawShape.isInFilled ? this.drawShape.color : '',
            stroke: this.drawShape.color,
            originX: 'center',
            originY: 'center',
            stopPinchZoom: true // something like this top stop pinch zoom / resize
        });
```
Or is there any work around to prevent resize?

### Current State

when doing pinch with 2 fingers is start resizing an object.

**Comment by ShaMan123** [comment]
Please open discussions for these Q&A
I suggest you provide a lot more context if you want people to understand and help

**Reply by Saqib92** [reply]
any help?

**Comment by Saqib92** [comment]
I am making Mobile/ Touch screen capable application using Ionic / Angular, when user Drag and Drop and Image from side bar to canvas, I want to stop pinch zoom but at the same time I want user to drag multiple objects, currently when user selects Image1 to drag and try to select image2 it starts pinch zooming / resizing image1.

**Comment by ShaMan123** [comment]
Fabric lacks a proper interaction state manager.
You will need to overridfe stuff and do it yourself

**Reply by ShaMan123** [reply]
I want to write one for fabric but I have about 70 open PRs, each of them more important than the other.

**Comment by asturur** [comment]
from your vague description it seems you are using drag and drop from html elements and that when using a second finger during drag you get a zoom interaction.

regardless any state interaction code, pinch to zoom is not a browser api issue and is an accessibility issue too. 

There is no code to pinch to zoom in fabric you hooked up yourself, when stard dragging an image you can remove it. If is the native viewport trigger code that runs you can disable with the viewport attributes for the browsers tha allow it. Zooming is a user accessibility concern that is why pinch to zoom on the whole page is delegted to the browser.

**Reply by Saqib92** [reply]
Not exactly zoom, I am facing resize issue. I want to disable pinch resizing on selected object / image on canvas.  I want users to use Controls to resize objects i.e mtr, tr, tl, etc.

**Reply by asturur** [reply]
what is pinch resizing? who wrote the code for it?

**Reply by Saqib92** [reply]
I don't know. and Don't even know where it come from. But when I try to move any object on canvas and put two fingers on canvas it starts making selected image bigger / smaller depends on movement.

**Reply by asturur** [reply]
Well when you figure that out we can help, as you see there are too many variables, if you don't know if the zooming comes from the embedded browser pinching or website code how can we provide any guidance?
